src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -971,6 +971,7 @@ def create_python_udf_or_sp(
     secure: bool = False,
     external_access_integrations: Optional[List[str]] = None,
     secrets: Optional[Dict[str, str]] = None,
+    immutable: bool = False,
```

Comment:
can you check if those are the two possible values for this? such that we don't plan to introduce more in the future at least in short term.

because I'm thinking of if there's a third value for this parameter, however boolean which only represents two is not enough.

in contrast, another way to do is introducing two parameters in your PR:

```
immutable: Optional[bool] = None
volatile: Optional[bool] = None
```

Reply:
the keyword is for determinism of the UDF, which there're are only two options.